[TAN-4064] Fix project review request for folder moderators

adessy opened the pull request:
# Changelog
## Fixed
- [TAN-4064] Fix "Request approval" button for folder moderators. If project reviews are enabled, the "Request approval" button is now shown to folder moderators for projects outside the folders they manage. For example, if a moderator manages one folder and one standalone project outside of that folder, they can publish any project within their folder without approval, but for the other project, project moderator rules apply and the moderators must request approval. If the project review are disabled, the "Publish" button is shown, but is disabled.

adessy committed:
[TAN-4064] Fix project review request for folder moderators

If project reviews are enabled, the "Request approval" button is now shown to folder moderators for projects outside the folders they manage. For example, if a moderator manages one folder and one standalone project outside of that folder, they can publish any project within their folder without approval, but for the other project, project moderator rules apply.

Otherwise, the "Publish" button is shown, but is disabled.

adessy committed:
[TAN-4064] Improve project-review flow copy

Commit:
Translations updated by CI (extract-intl)

brentguf reviewed (approved):
Looks good already. Some suggestions to further improve things: https://github.com/CitizenLabDotCo/citizenlab/pull/10758

adessy committed:
[TAN-4064] Always show "Publish" button for projects to admins (and similar roles),

Show the "Publish" button even if the project is pending approval, so that an admin can publish it without needing to approve it first.

Co-authored-by: Brent Guffens <brentguf@users.noreply.github.com>
